src/plugins/intel_cpu/src/nodes/mvn.cpp

```diff
@@ -2694,55 +2700,75 @@ void MVN::MVNJitExecutor::mvn_nspc(const uint8_t* src_data,
+                    variance_ptr[0] += variance_ptr[i];
                 }
                 if (mvnAttrs.epsMode_ == INSIDE_SQRT) {
-                    variance_buffer[0] = 1.F / sqrtf(variance_buffer[0] * size_inv + mvnAttrs.epsValue_);
+                    variance_ptr[0] = 1.F / sqrtf(variance_ptr[0] * size_inv + mvnAttrs.epsValue_);
                 } else if (mvnAttrs.epsMode_ == OUTSIDE_SQRT) {
-                    variance_buffer[0] = 1.F / (sqrtf(variance_buffer[0] * size_inv) + mvnAttrs.epsValue_);
+                    variance_ptr[0] = 1.F / (sqrtf(variance_ptr[0] * size_inv) + mvnAttrs.epsValue_);
                 }
             }
             worker(true, 2);
         } else {  // for per_channel
             float size_inv = 1.F / static_cast<float>(D * H * W);
             worker(false, 0);
-            for (size_t i = 1; i < threads_num; i++) {
+            for (size_t i = 1; i < w_threads; i++) {
                 for (size_t c = 0; c < C; c++) {
-                    mean_buffer[c] += mean_buffer[c + aux_buffer_size * i];
+                    mean_ptr[c] += mean_ptr[c + aux_buffer_size * i];
                 }
             }
             for (size_t c = 0; c < C; c++) {
-                mean_buffer[c] *= size_inv;
+                mean_ptr[c] *= size_inv;
             }
             if (mvnAttrs.normalizeVariance_) {
                 worker(false, 1);
-                for (size_t i = 1; i < threads_num; i++) {
+                for (size_t i = 1; i < w_threads; i++) {
                     for (size_t c = 0; c < C; c++) {
-                        variance_buffer[c] += variance_buffer[c + aux_buffer_size * i];
+                        variance_ptr[c] += variance_ptr[c + aux_buffer_size * i];
                     }
                 }
                 for (size_t c = 0; c < C; c++) {
                     if (mvnAttrs.epsMode_ == INSIDE_SQRT) {
-                        variance_buffer[c] = 1.F / sqrtf(variance_buffer[c] * size_inv + mvnAttrs.epsValue_);
+                        variance_ptr[c] = 1.F / sqrtf(variance_ptr[c] * size_inv + mvnAttrs.epsValue_);
                     } else if (mvnAttrs.epsMode_ == OUTSIDE_SQRT) {
-                        variance_buffer[c] = 1.F / (sqrtf(variance_buffer[c] * size_inv) + mvnAttrs.epsValue_);
+                        variance_ptr[c] = 1.F / (sqrtf(variance_ptr[c] * size_inv) + mvnAttrs.epsValue_);
                     }
                 }
             }
             worker(false, 2);
         }
     };
 
-    parallel_nt_static(threads_num, [&](const int ithr, const int nthr) {
+#if OV_THREAD == OV_THREAD_OMP
+    auto origin_nested_levels = parallel_get_max_nested_levels();
+    if (origin_nested_levels <= parallel_get_nested_level() + 1) {
+        parallel_set_max_nested_levels(origin_nested_levels + 2);
+    }
+    auto origin_nested = parallel_get_nested();
+    if (origin_nested == 0) {
+        parallel_set_nested(1);
+    }
+#endif
```

Comment:
Can we use a RAII structure instead of two calls (set nested/restore nested)? That would be more convenient and secure for exceptions

Reply:
Done

Reply:
RAII class can make the code even simpler and allow restore the nested context in case the exception is thrown:
```cpp
class ParallelNestingContext {
public:
    ParallelNestingContext() {
#if OV_THREAD == OV_THREAD_OMP
        m_state = parallel_enable_nesting();
#endif
    }
    ~ParallelNestingContext() {
#if OV_THREAD == OV_THREAD_OMP
        parallel_restore_nesting(m_state);
#endif
    }

    ParallelNestingContext(const ParallelNestingContext&) = delete;
    ParallelNestingContext& operator=(const ParallelNestingContext&) = delete;
private:
    int m_state{0};
};


//usage

    ParallelNestingContext  nested_context;
    parallel_nt_static(b_threads, [&](const int ithr, const int nthr) {
        for_1d(ithr, nthr, N, b_loop);
    });
```

Reply:
Done